api/test/test_play_asset_pack_manager.gd

```diff
@@ -163,11 +163,136 @@ func test_cancel_asset_pack_request_non_existing_pack_name():
+	var expected_pack_state_dict = create_mock_asset_pack_state_dict()
+	var expected_pack_name = expected_pack_state_dict[PlayAssetPackState._NAME_KEY]
+	assert_true(did_succeed)
+	assert_eq(pack_name, expected_pack_name)
+	assert_asset_pack_state_eq_dict(result, expected_pack_state_dict)
+	assert_eq(exception, null)
+
+func test_get_asset_pack_state_error():
+	var mock_plugin = FakeAndroidPlugin.new()
+	
+	var handler = FakeGetPackStatesHandler.new(false, {}, \
+		create_mock_asset_pack_java_lang_exception_dict())
+	mock_plugin.set_fake_get_pack_states_handler(handler)
+	
+	var test_object = create_play_asset_pack_manager(mock_plugin)
+	
+	var test_pack_name = "random pack name"
+	var request_object = test_object.get_asset_pack_state(test_pack_name)
+	
+	# connect to helper function, simulating the workflow of connecting callback to signal
+	request_object.connect("request_completed", self, "assert_get_asset_pack_state_signal_is_error")
+	
+	# yield to the request_completed signal for no longer than 1 seconds and assert for signal emitted
+	yield(yield_to(request_object, "request_completed", 1), YIELD)
+	assert_signal_emitted(request_object, "request_completed", "signal should have emitted")
+	
+	# assert using getters, simulating the workflow of yielding the signals
+	assert_true(not request_object.get_did_succeed())
+	assert_eq(request_object.get_pack_name(), test_pack_name)
+	assert_asset_pack_exception_eq_dict(request_object.get_error(), create_mock_asset_pack_java_lang_exception_dict())
+	assert_eq(request_object.get_result(), null)
+	
+	# join instantiated thread
+	handler.thread.wait_to_finish()
+
+func assert_get_asset_pack_state_signal_is_error(did_succeed : bool, pack_name : String, \
+	result : PlayAssetPackState, exception : PlayAssetPackException):
+	# assert using callback, simulating the workflow of connecting callback to signal
+	assert_true(not did_succeed)
+	assert_eq(pack_name, "random pack name")
+	assert_asset_pack_exception_eq_dict(exception, create_mock_asset_pack_java_lang_exception_dict())
+	assert_eq(result, null)
+
+func test_get_asset_pack_state_non_existent_pack():
+	var mock_plugin = FakeAndroidPlugin.new()
+
+	# configure what should be emitted upon get_asset_pack_state() call
+	var test_asset_pack_state = create_mock_asset_pack_state_dict()
+	var test_pack_name = test_asset_pack_state[PlayAssetPackState._NAME_KEY]
+	
+	# the plugin call will return an AssetPackStates dictionary enclose the given test_asset_pack_state
+	var test_asset_pack_states = create_mock_asset_pack_states_with_single_state_dict(test_asset_pack_state)
+	var handler = FakeGetPackStatesHandler.new(true, \
+		test_asset_pack_states, {})
+	mock_plugin.set_fake_get_pack_states_handler(handler)
+	
+	var test_object = create_play_asset_pack_manager(mock_plugin)
+	
+	# AssetPackStates dictionary returned by plugin should not contain this pack_name
+	var non_existent_pack_name = test_pack_name + "non-existent suffix"
```

Comment:
Do you need to derive the pack name from `test_pack_name `? Might simplify things a bit if you just name it "non_existent_pack".

Reply:
Yeah I think simplifying this logic is a good call. Renamed the pack name to ```non_existent_pack```.